File: libs/envelope-py/orchestack_envelope/idempotency.py
"""Idempotency key generation per RFC-001."""


def generate_idempotency_key(
    tenant_id: str,
    task_id: str,
    run_attempt: int,
    step_type: str,
    step_seq: int,
) -> str:
    """Generate idempotency key: idem:{tenant}:{task_id}:{run_attempt}:{step_type}:{step_seq}."""
    return f"idem:{tenant_id}:{task_id}:{run_attempt}:{step_type}:{step_seq}"

File: libs/envelope-ts/src/idempotency.ts
/**
 * Idempotency key generation per RFC-001.
 */
export function generateIdempotencyKey(
  tenantId: string,
  taskId: string,
  runAttempt: number,
  stepType: string,
  stepSeq: number,
): string {
  return `idem:${tenantId}:${taskId}:${runAttempt}:${stepType}:${stepSeq}`;
}
